src/as2fm/jani_generator/ros_helpers/ros_timer.py

```diff
@@ -244,27 +126,51 @@ def make_global_timer_scxml(timers: List[RosTimer], max_time_ns: int) -> Optiona
             f"Max time {max_time_ns}ns cannot be converted to '{global_timer_period_unit}'. "
             "The max_time must have a unit that is greater or equal to the smallest timer period."
         )
-    scxml_root = ScxmlRoot("global_timer_automata")
-    scxml_root.set_data_model(ScxmlDataModel([ScxmlData("current_time", "0", "int64")]))
+    curr_time_var = "current_time"
+    scxml_root = ScxmlRoot(GLOBAL_TIMER_AUTOMATON)
+    clock_topic_decl = RosTopicPublisher("clock", "builtin_interfaces/Time")
+    scxml_root.add_ros_declaration(clock_topic_decl)
+    scxml_root.set_data_model(ScxmlDataModel([ScxmlData(curr_time_var, "0", "int64")]))
     idle_state = ScxmlState("idle")
-    global_timer_tick_body: ScxmlExecutionBody = []
-    global_timer_tick_body.append(
-        ScxmlAssign("current_time", f"current_time + {global_timer_period}")
-    )
+    global_timer_tick_body: ScxmlExecutionBody = [
```

Comment:
When ticking the internal time, we need to tell the SCXML model that some "actual" time has to pass. I propose to do this done via sending an event with a delay:

```xml
<send event="TIME" target="NONE" delay="1"/>
```

(so that the internal time unit corresponds to the global timer period).

There might be better names for the event, feel free to rename it. I am unsure if this event should be paired with a ROS topic, but it is probably unnecessary.

I think what would be required is to add something like the line:

```py
ScxmlSend(f"{"TIME"}{"NONE"}")
```

Does this sound right to you? Is it fine to do this modification?

Reply:
I see your point, but this particular event will be a "plain scxml" event, with no relation to a specific time unit, unlike what we are providing using the ROS message.
Would you be fine to use that? providing we add the target and the delay automatically?